lib/vscode/src/vs/server/node/channel.ts

```diff
@@ -502,7 +500,7 @@ export class TerminalProviderChannel implements IServerChannel<RemoteAgentConnec
 		};
 
 		const getEnvironment = async (): Promise<platform.IProcessEnvironment> => {
-			const env = await this.terminalProfileResolverService.getEnvironment(remoteAuthority);
+			const env = await this.ptyService.getEnvironment();
 			env.VSCODE_IPC_HOOK_CLI = process.env['VSCODE_IPC_HOOK_CLI']!;
 			return env;
 		};
```

Comment:
We can get rid of this wrapper function and call `this.ptyService.getEnvironment()` directly.

Previously this fetched the parent's environment which is why we had to add `VSCODE_IPC_HOOK_CLI` back but now that it always inherits the environment we don't need to set it manually.

Actually I think maybe we just use `process.env` directly since that's what we had before when `inheritEnv` was optional.

Reply:
Discussed elsewhere, but conclusion is we're keeping it this way:
ptyHostService calls getEnvironment in the ptyHost process it creates, which uses that process's environment. The process spawned doesn't have VSCODE_IPC_HOOK_CLI in its env, so we add it here.